src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -177,9 +178,17 @@ def _is_blocked(obj: Union[Task, Job]) -> bool:
         Returns:
              True if one of its input data nodes is blocked.
         """
+
         data_nodes = obj.task.input.values() if isinstance(obj, Job) else obj.input.values()
         data_manager = _DataManagerFactory._build_manager()
-        return any(not data_manager._get(dn.id).is_ready_for_reading for dn in data_nodes)
+
+        def get_dn_is_ready_for_reading(dn: DataNode):
+            dn = data_manager._get(dn.id)
+            if dn.is_ready_for_reading is None:
+                cls.__logger.error(f"{dn.id} is not ready to be read.")
+            return dn.is_ready_for_reading
+
+        return any(not get_dn_is_ready_for_reading(dn) for dn in data_nodes)
```

Comment:
`dn.is_ready_for_reading` can't be None. It is either True or False. I believe the msg is never logged.

Plus, sometimes, it is normal to have a blocked job. It is not an error but a warning.
I would go for a more precise msg. Maybe we can have something like that. What do you think?

```
        def get_dn_is_ready_for_reading(dn: DataNode):
            dn = data_manager._get(dn.id)
            if dn.is_ready_for_reading is False:
                if not dn._last_edit_date:
                    if dn.storage_type() == "csv' or 'excel' or any other file
                        cls.__logger.warning(f"{dn.id} cannot be read because it has never been written. Hint: The data node may refer to a wrong path : {dn.path}")
                    else:
                        cls.__logger.warning(f"{dn.id} cannot be read because it has never been written.")
            return dn.is_ready_for_reading

        return any(not get_dn_is_ready_for_reading(dn) for dn in data_nodes)
```